atd-vze/src/Components/GridTable.js

```diff
@@ -0,0 +1,226 @@
+   *      offset {int} - Contains the current offset for querying in the database
+   *      page {int} - Contains the current page number (1 by default)
+   *      sortColumn {string} - Contains the name of the column being used to sort the data
+   *      sortOrder {string} - Contains either 'asc' or 'desc'
+   *      searchParameters {object} - Contains the parameters for the text search
+   */
+  const [limit, setLimit] = useState(25);
+  const [offset, setOffset] = useState(0);
+  const [page, setPage] = useState(1);
+  const [sortColumn, setSortColumn] = useState("");
+  const [sortOrder, setSortOrder] = useState("");
+  const [searchParameters, setSearchParameters] = useState({});
+
+  /**
+   * Handles the header click for sorting asc/desc.
+   * @param {string} col - The name of the column
+   **/
+  const handleTableHeaderClick = col => {
+    if (sortOrder === "" && sortColumn === "") {
+      // First time sort is applied
+      setSortOrder("asc");
+      setSortColumn(col);
+    } else if (sortColumn === col) {
+      // Repeat sort on column
+      sortOrder === "desc" ? setSortOrder("asc") : setSortOrder("desc");
+    } else if (sortColumn !== col) {
+      // Sort different column after initial sort
+      setSortOrder("desc");
+      setSortColumn(col);
+    }
+  };
+
+  /**
+   * Changes the state of the current page, and offset limit.
+   * @param {integer} n - the page we need to change it to
+   */
+  const changePage = n => {
+    console.log("Changing page to: " + n);
+    setPage(n);
+    setOffset(n * limit - limit);
+  };
+
+  /**
+   * Moves to the next page of results, by changing the state to current page plus one.
+   */
+  const moveNextPage = () => {
+    changePage(page + 1);
+  };
+
+  /**
+   * Moves to the previous page of results, by substracting 1 to the current page.
+   */
+  const moveBackPage = () => {
+    changePage(page - 1);
+  };
+
+  /**
+   * Handles a click on the limit dropdown menu.
+   * @param {object} e - the event parameter
+   */
+  const handleRowClick = e => {
+    const rowNumber = parseInt(e.target.value);
+    setLimit(rowNumber);
+  };
+
+  /**
+   * Clears all filters
+   */
+  const clearFilters = () => {
+    setSearchParameters({});
+  };
+
+  /**
+   *
+   * Query Management
+   *
+   **/
+
+  // Manage the WHERE clause of our query
+  query.cleanWhere(); // Clean slate
+  if (searchParameters["column"] && searchParameters["value"]) {
+    query.setWhere(
+      searchParameters["column"],
+      `_like: "%${searchParameters["value"]}%"`
+    );
+  }
+
+  // Manage the ORDER BY clause of our query
+  if (sortColumn !== "" && sortOrder !== "") {
+    console.log("Changing order...");
+    query.setOrder(sortColumn, sortOrder);
+  }
+
+  // Mange LIMIT & OFFSET
+  query.limit = limit;
+  query.offset = offset;
+
+  /**
+   *
+   * Render
+   *
+   **/
+
+  // Make Query && Error handling
+  console.log(query.query);
```

Comment:
https://www.quora.com/Is-it-bad-form-to-leave-console.log()-lines-in-your-JavaScript